Libraries/Opc.Ua.PubSub/Encoding/UadpNetworkMessage.cs

```diff
@@ -827,6 +919,38 @@ private void DecodeMetaDataMessage(BinaryDecoder binaryDecoder)
 
         }
 
+        /// <summary>
+        /// Decode the binaryDecoder content as Endpoints message
+        /// </summary>
+        /// <param name="binaryDecoder"></param>
+        private void DecodePublisherEndpoints(BinaryDecoder binaryDecoder)
+        {
+            PublisherEndpoints = (EndpointDescription[])binaryDecoder.ReadEncodeableArray("Endpoints", typeof(EndpointDescription));
+
+            PublisherProvideEndpoints = binaryDecoder.ReadStatusCode("statusCode");
+
+            Trace("DecodePublisherEndpointsMessage returned: ", PublisherProvideEndpoints);
+        }
+
+        /// <summary>
+        /// Decode the binaryDecoder content as a DataSetWriterConfiguration message
+        /// </summary>
+        /// <param name="binaryDecoder">the decoder</param>
+        private void DecodeDataSetWriterConfigurationMessage(BinaryDecoder binaryDecoder)
+        {
+            DataSetWriterIds = binaryDecoder.ReadUInt16Array("DataSetWriterIds").ToArray();
+
+            var dataSetWriterConfigurationDecoded = binaryDecoder.ReadEncodeable("DataSetWriterConfiguration", typeof(WriterGroupDataType)) as WriterGroupDataType;
+
+            DataSetWriterConfiguration = dataSetWriterConfigurationDecoded.MaxNetworkMessageSize != 0
+                ? dataSetWriterConfigurationDecoded
+                : null;
+
+            // temporary write StatusCode.Good 
+            MessageStatusCodes = binaryDecoder.ReadStatusCodeArray("StatusCodes").ToArray();
+            Trace("DecodeDataSetWriterConfigurationMessage returned: ", MessageStatusCodes);
```

Comment:
please use the Utils.LogTrace methods. The trace calls may be deprecated soon.

Reply:
Looks like the pubsub sample is still using the older Utils.Trace across the board. We need a work item to convert these, as they all map to Utils.LogInfo.